modules/oe_multilingual_url_suffix/src/Plugin/LanguageNegotiation/LanguageNegotiationUrlSuffix.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class LanguageNegotiationUrlSuffix.
+ *
+ * @LanguageNegotiation(
+ *   id = \Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation\LanguageNegotiationUrlSuffix::METHOD_ID,
+ *   types = {\Drupal\Core\Language\LanguageInterface::TYPE_INTERFACE,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_CONTENT,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_URL},
+ *   weight = -10,
+ *   name = @Translation("URL suffix"),
+ *   description = @Translation("Language from the URL (Path suffix)."),
+ *   config_route_name = "oe_multilingual_url_suffix.negotiation_url_suffix"
+ * )
+ */
+class LanguageNegotiationUrlSuffix extends LanguageNegotiationUrl {
+
+  /**
+   * The language negotiation method id.
+   */
+  const METHOD_ID = 'language-url-suffix';
+
+  /**
+   * The suffix delimiter.
+   */
+  const SUFFIX_DELIMITER = '_';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLangcode(Request $request = NULL) {
+    $langcode = NULL;
+
+    $config = $this->config->get('language.negotiation')->get('url_suffixes');
```

Comment:
This will need to be changed with our own config object.

Reply:
Done